src/test/java/bootstrap/DriverTest.java

```diff
@@ -0,0 +1,108 @@
+package bootstrap;
+
+import domain.Astar;
+import domain.Cell;
+import domain.CellComparator;
+import domain.GenerateMaze;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.*;
+
+public class DriverTest {
+
+    @Test
+    public void ValidPath() {
+        List<Integer> wallsList = new ArrayList<>();
+        String arg = "--log.file.path executionLogs --log.level INFO --difficulty easy --start 3 --dest 58 --size 10";
+        String[] args = arg.split(" ");
+        GenerateMaze gmaze = new GenerateMaze(args);
+        int[][] maze = gmaze.getMaze();
+        Integer[] walls = new Integer[]{81, 82, 51, 83, 52, 53, 93};
+        wallsList.addAll(Arrays.asList(walls));
+        int destId = Integer.parseInt(Driver.getArgument(args, "dest"));
+        Astar run = new Astar(maze, args, Arrays.asList(walls));
+        Cell Result = run.findPath();
+        Assert.assertEquals(destId, Result.getId());
+    }
+
+    @Test
+    public void InvalidPath() {
+        List<Integer> wallsList = new ArrayList<>();
+        String arg = "--log.file.path executionLogs --log.level INFO --difficulty easy --start 3 --dest 58 --size 10";
+        String[] args = arg.split(" ");
+        GenerateMaze gmaze = new GenerateMaze(args);
+        int[][] maze = gmaze.getMaze();
+        Integer[] walls = new Integer[]{49,48,47,57,67,68,69};
+        wallsList.addAll(Arrays.asList(walls));
+        int destId = Integer.parseInt(Driver.getArgument(args, "dest"));
+        Astar run = new Astar(maze, args, Arrays.asList(walls));
+        Cell Result = run.findPath();
+        Assert.assertEquals(null, Result);
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testInvalidArgs() {
+        String arg = "--log.file.path executionLogs --log.level INFO --difficulty easy --start 3 --dest 58 --size";
+        String[] args = arg.split(" ");
+        int size = Integer.parseInt(Driver.getArgument(args, "size"));
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testInvalidArgs1() {
```

Comment:
rename this function using numbers at end to get different function names is never acceptable.